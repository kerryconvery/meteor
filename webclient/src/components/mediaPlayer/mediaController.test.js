import React from 'react';
import { shallow } from 'enzyme';
import MediaController from './mediaController';

describe('MediaController', () => {
  it('should match snapshot when playing', () => {
    const wrapper = shallow(<MediaController
      nowPlaying='movie.avi'
      playerState={2}
      position={100}
      duration={1000}
      onRestart={() => {}}
      onPause={() => {}}
      onStop={() => {}}
      onResume={() => {}}
      onPark={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when paused', () => {
    const wrapper = shallow(<MediaController
      nowPlaying='movie.avi'
      playerState={1}
      position={100}
      duration={1000}
      onRestart={() => {}}
      onPause={() => {}}
      onStop={() => {}}
      onResume={() => {}}
      onPark={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
