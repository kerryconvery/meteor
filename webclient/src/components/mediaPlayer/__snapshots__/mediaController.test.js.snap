// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaController should match snapshot when paused 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MediaController
    duration={1000}
    nowPlaying="movie.avi"
    onPark={[Function]}
    onPause={[Function]}
    onRestart={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playerState={1}
    position={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        style={
          Object {
            "marginLeft": "10px",
            "marginRight": "10px",
          }
        }
      >
        <span>
          Now Playing: 
          movie.avi
        </span>
        <Progress
          max={1000}
          tag="div"
          value={100}
        />
        <ButtonGroup
          className="d-flex justify-content-sm-center"
          role="group"
          tag="div"
        >
          <Button
            color="info"
            onClick={[Function]}
            tag="button"
          >
            Restart
          </Button>
          <Button
            color="info"
            onClick={[Function]}
            tag="button"
          >
            Resume
          </Button>
          <Button
            color="info"
            onClick={[Function]}
            tag="button"
          >
            Stop
          </Button>
          <Button
            color="info"
            onClick={[Function]}
            tag="button"
          >
            Park
          </Button>
        </ButtonGroup>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Now Playing: 
            movie.avi
          </span>,
          <Progress
            max={1000}
            tag="div"
            value={100}
          />,
          <ButtonGroup
            className="d-flex justify-content-sm-center"
            role="group"
            tag="div"
          >
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Restart
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Resume
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Stop
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Park
            </Button>
          </ButtonGroup>,
        ],
        "style": Object {
          "marginLeft": "10px",
          "marginRight": "10px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Now Playing: ",
              "movie.avi",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Now Playing: ",
            "movie.avi",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "max": 1000,
            "tag": "div",
            "value": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Restart
              </Button>,
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Resume
              </Button>,
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Stop
              </Button>,
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Park
              </Button>,
            ],
            "className": "d-flex justify-content-sm-center",
            "role": "group",
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Restart",
                "color": "info",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Restart",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Resume",
                "color": "info",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Resume",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Stop",
                "color": "info",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Stop",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Park",
                "color": "info",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Park",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          style={
            Object {
              "marginLeft": "10px",
              "marginRight": "10px",
            }
          }
        >
          <span>
            Now Playing: 
            movie.avi
          </span>
          <Progress
            max={1000}
            tag="div"
            value={100}
          />
          <ButtonGroup
            className="d-flex justify-content-sm-center"
            role="group"
            tag="div"
          >
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Restart
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Resume
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Stop
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Park
            </Button>
          </ButtonGroup>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Now Playing: 
              movie.avi
            </span>,
            <Progress
              max={1000}
              tag="div"
              value={100}
            />,
            <ButtonGroup
              className="d-flex justify-content-sm-center"
              role="group"
              tag="div"
            >
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Restart
              </Button>
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Resume
              </Button>
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Stop
              </Button>
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Park
              </Button>
            </ButtonGroup>,
          ],
          "style": Object {
            "marginLeft": "10px",
            "marginRight": "10px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Now Playing: ",
                "movie.avi",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Now Playing: ",
              "movie.avi",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "max": 1000,
              "tag": "div",
              "value": 100,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Button
                  color="info"
                  onClick={[Function]}
                  tag="button"
                >
                  Restart
                </Button>,
                <Button
                  color="info"
                  onClick={[Function]}
                  tag="button"
                >
                  Resume
                </Button>,
                <Button
                  color="info"
                  onClick={[Function]}
                  tag="button"
                >
                  Stop
                </Button>,
                <Button
                  color="info"
                  onClick={[Function]}
                  tag="button"
                >
                  Park
                </Button>,
              ],
              "className": "d-flex justify-content-sm-center",
              "role": "group",
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Restart",
                  "color": "info",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Restart",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Resume",
                  "color": "info",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Resume",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Stop",
                  "color": "info",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Stop",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Park",
                  "color": "info",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Park",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MediaController should match snapshot when playing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MediaController
    duration={1000}
    nowPlaying="movie.avi"
    onPark={[Function]}
    onPause={[Function]}
    onRestart={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playerState={2}
    position={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        style={
          Object {
            "marginLeft": "10px",
            "marginRight": "10px",
          }
        }
      >
        <span>
          Now Playing: 
          movie.avi
        </span>
        <Progress
          max={1000}
          tag="div"
          value={100}
        />
        <ButtonGroup
          className="d-flex justify-content-sm-center"
          role="group"
          tag="div"
        >
          <Button
            color="info"
            onClick={[Function]}
            tag="button"
          >
            Restart
          </Button>
          <Button
            color="info"
            onClick={[Function]}
            tag="button"
          >
            Pause
          </Button>
          <Button
            color="info"
            onClick={[Function]}
            tag="button"
          >
            Stop
          </Button>
          <Button
            color="info"
            onClick={[Function]}
            tag="button"
          >
            Park
          </Button>
        </ButtonGroup>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Now Playing: 
            movie.avi
          </span>,
          <Progress
            max={1000}
            tag="div"
            value={100}
          />,
          <ButtonGroup
            className="d-flex justify-content-sm-center"
            role="group"
            tag="div"
          >
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Restart
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Pause
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Stop
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Park
            </Button>
          </ButtonGroup>,
        ],
        "style": Object {
          "marginLeft": "10px",
          "marginRight": "10px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Now Playing: ",
              "movie.avi",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Now Playing: ",
            "movie.avi",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "max": 1000,
            "tag": "div",
            "value": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Restart
              </Button>,
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Pause
              </Button>,
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Stop
              </Button>,
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Park
              </Button>,
            ],
            "className": "d-flex justify-content-sm-center",
            "role": "group",
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Restart",
                "color": "info",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Restart",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Pause",
                "color": "info",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Pause",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Stop",
                "color": "info",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Stop",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Park",
                "color": "info",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Park",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          style={
            Object {
              "marginLeft": "10px",
              "marginRight": "10px",
            }
          }
        >
          <span>
            Now Playing: 
            movie.avi
          </span>
          <Progress
            max={1000}
            tag="div"
            value={100}
          />
          <ButtonGroup
            className="d-flex justify-content-sm-center"
            role="group"
            tag="div"
          >
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Restart
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Pause
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Stop
            </Button>
            <Button
              color="info"
              onClick={[Function]}
              tag="button"
            >
              Park
            </Button>
          </ButtonGroup>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Now Playing: 
              movie.avi
            </span>,
            <Progress
              max={1000}
              tag="div"
              value={100}
            />,
            <ButtonGroup
              className="d-flex justify-content-sm-center"
              role="group"
              tag="div"
            >
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Restart
              </Button>
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Pause
              </Button>
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Stop
              </Button>
              <Button
                color="info"
                onClick={[Function]}
                tag="button"
              >
                Park
              </Button>
            </ButtonGroup>,
          ],
          "style": Object {
            "marginLeft": "10px",
            "marginRight": "10px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Now Playing: ",
                "movie.avi",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Now Playing: ",
              "movie.avi",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "max": 1000,
              "tag": "div",
              "value": 100,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Button
                  color="info"
                  onClick={[Function]}
                  tag="button"
                >
                  Restart
                </Button>,
                <Button
                  color="info"
                  onClick={[Function]}
                  tag="button"
                >
                  Pause
                </Button>,
                <Button
                  color="info"
                  onClick={[Function]}
                  tag="button"
                >
                  Stop
                </Button>,
                <Button
                  color="info"
                  onClick={[Function]}
                  tag="button"
                >
                  Park
                </Button>,
              ],
              "className": "d-flex justify-content-sm-center",
              "role": "group",
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Restart",
                  "color": "info",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Restart",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Pause",
                  "color": "info",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Pause",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Stop",
                  "color": "info",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Stop",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Park",
                  "color": "info",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Park",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
